geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/key/RenameDUnitTest.java

```diff
@@ -210,6 +227,48 @@ private void doConcurrentRenames(List<String> listOfKeys1, List<String> listOfKe
     future4.get();
   }
 
+  @Test
+  public void givenCrashDuringRename_thenDoesNotLeaveInconsistencies() throws Exception {
+    final AtomicBoolean running = new AtomicBoolean(true);
+
+    final List<String> hashtags = new ArrayList<>();
+    hashtags.add(clusterStartUp.getKeyOnServer("rename", 1));
+    hashtags.add(clusterStartUp.getKeyOnServer("rename", 2));
+    hashtags.add(clusterStartUp.getKeyOnServer("rename", 3));
+
+    final Runnable task1 = () -> renamePerformAndVerify(1, 10000, hashtags.get(0), running, true);
+    final Runnable task2 = () -> renamePerformAndVerify(2, 10000, hashtags.get(1), running, true);
+    final Runnable task3 = () -> renamePerformAndVerify(3, 10000, hashtags.get(2), running, true);
+
+    final Future<Void> future1 = executor.runAsync(task1);
+    final Future<Void> future2 = executor.runAsync(task2);
+    final Future<Void> future3 = executor.runAsync(task3);
+
+    final String finalRedisPort = Integer.toString(server3Port);
+    final int finalLocatorPort = locatorPort;
+    final Future<?> crasherFuture = executor.submit(() -> {
+      try {
+        for (int i = 0; i < 20 && running.get(); i++) {
+          clusterStartUp.moveBucketForKey(hashtags.get(0), "server-3");
```

Comment:
Is this necessary? It seems like crashing the server should be sufficient to trigger the bug, without needing to move the bucket first.

Reply:
This follows the pattern in MSetDUnitTest `testMSet_crashDoesNotLeaveInconsistencies` which Jens wrote. I believe the purpose of this line to make sure there are keys hosted on the server that is about to be crashed.

Reply:
Perhaps when the server has previously crashed there may not be keys hosted there anymore. Maybe Jens can comment on this as well if there's more to it than this.